src/orchestrator.rs

```diff
@@ -238,6 +238,14 @@ async fn detect(
             let ctx = ctx.clone();
             let detector_id = detector_id.clone();
             let detector_params = detector_params.clone();
+            // Get the default threshold to use if threshold is not provided by the user
```

Comment:
Now that we need two items from the detector config, I would just `get` the detector config once (returning an error if it's not found) and get `default_threshold` and `chunker_id` from it. One less lookup and then you don't have to do `.ok_or_else(|| Error::DetectorNotFound {}))` twice.

You can also drop that `OrchestratorConfig::get_default_threshold()` method then.

e.g.
```rust
// Get the detector config
let detector_config =
    ctx.config
        .detectors
        .get(&detector_id)
        .ok_or_else(|| Error::DetectorNotFound {
            detector_id: detector_id.clone(),
        })?;
// Get the default threshold to use if threshold is not provided by the user
let default_threshold = detector_config.default_threshold;
// Get chunker for detector
let chunker_id = detector_config.chunker_id.as_str();
let chunks = chunks.get(chunker_id).unwrap().clone();
```